drf_renderer_xlsx/fields.py

```diff
@@ -0,0 +1,185 @@
+	def prep_value(self) -> Any:
+		return self.value
+
+	def prep_cell(self, cell: Cell):
+		# We cannot apply the whole style directly, otherwise we cannot override any part of it
+		cell.font = self.style.font
+		cell.fill = self.style.fill
+		cell.alignment = self.style.alignment
+		cell.border = self.style.border
+		cell.number_format = self.style.number_format
+
+	def cell(self, ws: Worksheet, row, column) -> Cell:
+		# If we have a custom mapping use it and done. If not prep value for output
+		value = self.sanitize_value(self.custom_mapping() if self.mapping else self.prep_value())
+		cell: Cell = ws.cell(row, column, value)
+		self.prep_cell(cell)
+		return cell
+
+
+class XLSXNumberField(XLSXField):
+	sanitize = False
+
+	def __init__(self, number_format, **kwargs):
+		self.number_format = number_format
+		super().__init__(**kwargs)
+
+	def init_value(self, value):
+		try:
+			if isinstance(self.drf_field, IntegerField) and type(value) != int:
+				return int(value)
+			elif isinstance(self.drf_field, FloatField) and type(value) != float:
+				return float(value)
+			elif isinstance(self.drf_field, DecimalField) and type(value) != Decimal:
+				return Decimal(value)
+		except:
+			pass
+		return value
+
+	def prep_cell(self, cell: Cell):
+		super().prep_cell(cell)
+		if self.number_format:
+			cell.number_format = self.number_format
+		elif isinstance(self.drf_field, IntegerField):
+			cell.number_format = get_setting('INTEGER_FORMAT') or FORMAT_NUMBER
+		else:
+			cell.number_format = get_setting('DECIMAL_FORMAT') or FORMAT_NUMBER_00
+
+
+class XLSXDateField(XLSXField):
+	sanitize = False
+
+	def __init__(self, date_format, **kwargs):
+		self.date_format = date_format
+		super().__init__(**kwargs)
+
+	def _parse_date(self, value, setting_format, iso_parse_func):
+		# Parse format is DRF output format: DATETIME_FORMAT, DATE_FORMAT or TIME_FORMAT
+		parse_format = getattr(drf_settings, setting_format)
+		# Use the provided iso parse function for the special case ISO_8601
+		if parse_format.lower() == ISO_8601:
+			return iso_parse_func(value)
+		else:
+			parsed_datetime = datetime.datetime.strptime(value, parse_format)
+			if isinstance(self.drf_field, TimeField):
+				return parsed_datetime.time()
+			elif isinstance(self.drf_field, DateField):
+				return parsed_datetime.date()
+			return parsed_datetime
+
+	def init_value(self, value):
+		try:
+			if isinstance(self.drf_field, DateTimeField) and type(value) != datetime.datetime:
+				return self._parse_date(value, 'DATETIME_FORMAT', parse_datetime)
+			elif isinstance(self.drf_field, DateField) and type(value) != datetime.date:
+				return self._parse_date(value, 'DATE_FORMAT', parse_date)
+			elif isinstance(self.drf_field, TimeField) and type(value) != datetime.time:
+				return self._parse_date(value, 'TIME_FORMAT', parse_time)
+		except:
+			return value
+
+	def prep_cell(self, cell: Cell):
+		super().prep_cell(cell)
+		if self.date_format:
+			cell.number_format = self.date_format
```

Comment:
doesn't support stftime formatting. Current implementations may break. e.g. we currently use:
```
xlsx_date_format_mappings = {
        'created_at': '%d.%m.%Y %H:%M',
}
```

This seems to. be a "problem" with openpyxl i guess. I would be okay with marking this as a breaking change

Reply:
yeah, the formatting from now on would have to follow openpyxl for dates, numbers, etc
https://openpyxl.readthedocs.io/en/stable/_modules/openpyxl/styles/numbers.html
another option would be to set a flag and allow both (strftime would then be the default) but then the cell would have to be formatted as text (probably not a big deal)

Reply:
done. I use black already but in another project using tabs so it was all messed up. sorry about that.